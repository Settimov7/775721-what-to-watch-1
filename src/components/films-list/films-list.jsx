import * as React from 'react';
import * as PropTypes from 'prop-types';
import {connect} from "react-redux";

import {FilmCard} from "../film-card/film-card";
import {getFilteredFilms} from "../../reducer/films/selectors";

export const FilmsList = (props) => {
  const {films} = props;

  return (
    <div
      className="catalog__movies-list"
    >
      {films.map((film) => <FilmCard
        key={film.id}
        film={film}
      />)}
    </div>
  );
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  films: getFilteredFilms(state),
});

export default connect(mapStateToProps)(FilmsList);

FilmsList.propTypes = {
  films: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    posterImageSrc: PropTypes.string,
    previewImageSrc: PropTypes.string,
    backgroundImageSrc: PropTypes.string,
    backgroundColor: PropTypes.string,
    description: PropTypes.string,
    rating: PropTypes.number,
    scores: PropTypes.number,
    director: PropTypes.string,
    starring: PropTypes.arrayOf(PropTypes.string),
    runTime: PropTypes.number,
    genre: PropTypes.string,
    releasedYear: PropTypes.number,
    isFavorite: PropTypes.bool,
    videoSrc: PropTypes.string,
    previewVideoSrc: PropTypes.string,
  })).isRequired,
};
