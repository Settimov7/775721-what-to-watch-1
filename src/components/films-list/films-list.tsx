import * as React from 'react';
import {connect} from 'react-redux';

import {FilmCard} from '../film-card/film-card';
import {getFilteredFilms} from '../../reducer/films/selectors';
import {Film} from '../../types';

interface Props {
  films: Film[],
}

export const FilmsList = (props: Props) => {
  const {films} = props;

  return (
    <div
      className="catalog__movies-list"
    >
      {films.map((film) => <FilmCard
        key={film.id}
        film={film}
      />)}
    </div>
  );
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  films: getFilteredFilms(state),
});

export default connect(mapStateToProps)(FilmsList);
