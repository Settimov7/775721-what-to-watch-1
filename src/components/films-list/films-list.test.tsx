import * as React from 'react';
import * as renderer from 'react-test-renderer';

import {FilmsList} from "./films-list";

const mock = {
  films: [
    {
      id: 1,
      name: `Film-1`,
      posterImageSrc: `/image.jpg`,
      previewImageSrc: `/image.jpg`,
      backgroundImageSrc: `/image.jpg`,
      backgroundColor: `#123456`,
      description: `description`,
      rating: 5,
      scores: 123456,
      director: `director`,
      starring: [`person-1`, `person-2`, `person-3`],
      runTime: 99,
      genre: `Genre`,
      releasedYear: 2019,
      isFavorite: false,
      videoSrc: `video.mp4`,
      previewVideoSrc: `video.mp4`,
    },
    {
      id: 2,
      name: `Film-2`,
      posterImageSrc: `/image.jpg`,
      previewImageSrc: `/image.jpg`,
      backgroundImageSrc: `/image.jpg`,
      backgroundColor: `#123456`,
      description: `description`,
      rating: 5,
      scores: 123456,
      director: `director`,
      starring: [`person-1`, `person-2`, `person-3`],
      runTime: 99,
      genre: `Genre`,
      releasedYear: 2019,
      isFavorite: false,
      videoSrc: `video.mp4`,
      previewVideoSrc: `video.mp4`,
    },
    {
      id: 3,
      name: `Film-3`,
      posterImageSrc: `/image.jpg`,
      previewImageSrc: `/image.jpg`,
      backgroundImageSrc: `/image.jpg`,
      backgroundColor: `#123456`,
      description: `description`,
      rating: 5,
      scores: 123456,
      director: `director`,
      starring: [`person-1`, `person-2`, `person-3`],
      runTime: 99,
      genre: `Genre`,
      releasedYear: 2019,
      isFavorite: false,
      videoSrc: `video.mp4`,
      previewVideoSrc: `video.mp4`,
    },
    {
      id: 4,
      name: `Film-4`,
      posterImageSrc: `/image.jpg`,
      previewImageSrc: `/image.jpg`,
      backgroundImageSrc: `/image.jpg`,
      backgroundColor: `#123456`,
      description: `description`,
      rating: 5,
      scores: 123456,
      director: `director`,
      starring: [`person-1`, `person-2`, `person-3`],
      runTime: 99,
      genre: `Genre`,
      releasedYear: 2019,
      isFavorite: false,
      videoSrc: `video.mp4`,
      previewVideoSrc: `video.mp4`,
    },
    {
      id: 5,
      name: `Film-5`,
      posterImageSrc: `/image.jpg`,
      previewImageSrc: `/image.jpg`,
      backgroundImageSrc: `/image.jpg`,
      backgroundColor: `#123456`,
      description: `description`,
      rating: 5,
      scores: 123456,
      director: `director`,
      starring: [`person-1`, `person-2`, `person-3`],
      runTime: 99,
      genre: `Genre`,
      releasedYear: 2019,
      isFavorite: false,
      videoSrc: `video.mp4`,
      previewVideoSrc: `video.mp4`,
    },
  ],
};

it(`Films list correctly renders`, () => {
  const props = {
    films: mock.films,
  };

  const tree = renderer.create(<FilmsList {...props}/>).toJSON();

  expect(tree).toMatchSnapshot();
});
