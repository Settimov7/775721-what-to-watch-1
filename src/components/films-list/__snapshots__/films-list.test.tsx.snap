// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Films list correctly renders 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height={175}
        loop={true}
        muted={true}
        poster="/image.jpg"
        src="video.mp4"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Film-1
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height={175}
        loop={true}
        muted={true}
        poster="/image.jpg"
        src="video.mp4"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Film-2
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height={175}
        loop={true}
        muted={true}
        poster="/image.jpg"
        src="video.mp4"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Film-3
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height={175}
        loop={true}
        muted={true}
        poster="/image.jpg"
        src="video.mp4"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Film-4
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height={175}
        loop={true}
        muted={true}
        poster="/image.jpg"
        src="video.mp4"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Film-5
      </a>
    </h3>
  </article>
</div>
`;
